src/main/java/edu/umich/its/spe/SPEEsbImpl.java

```diff
@@ -26,28 +26,87 @@
 @Component
 @Qualifier("ESBIO")
 
+
 public class SPEEsbImpl implements GradeIO {
 
+	private static final String REUSE_CONNECTION = "reuseConnection";
+
 	protected static final String SKIP_GRADE_UPDATE = "SKIP GRADE UPDATE";
 
 	static final Logger M_log = LoggerFactory.getLogger(SPEEsbImpl.class);
 
 	@Autowired
 	SPESummary spesummary;
 
+	// Hold configuration for connecting to the ESB.
+	// If there are multiple WAPI connections required then will need multiple variables.
+	private WAPI wapi = null;
+
+	// Mostly for testing.  Default value set in assureWAPI below.
+	private Boolean reuseConnection = null;
+
 	// No arg constructor for use in SpringApp.
 	public SPEEsbImpl() {
 		super();
 	}
 
-	// Constructor that allows injecting summary object.  That can
-	// be useful in testing when don't want to require using Spring injection.
+	// Constructors that allow injecting objects.  This can be useful
+	// in testing when don't want to require using Spring injection.
+	// Only the specific constructors has been created.  There is probably a
+	// better way to do this.  The fundamental issue was injecting mocks.
 
 	public SPEEsbImpl(SPESummary spesummary) {
 		super();
 		this.spesummary = spesummary;
 	}
 
+	public SPEEsbImpl(WAPI wapi, SPESummary spesummary) {
+		super();
+		this.wapi = wapi;
+		this.spesummary = spesummary;
+	}
+
+	// allows testing default value of reuseConnection.
+	public SPEEsbImpl(WAPI wapi) {
+		super();
+		this.wapi = wapi;
+	}
+
+	public SPEEsbImpl(Boolean reuseConnection) {
+		super();
+		this.reuseConnection = reuseConnection;
+	}
+
+	// Make sure there is a useful connector to the ESB.  Pass in the value of caller WAPI by reference
+	// so a) can check if it is already set and b) this could be used for multiple different connections if necessary.
+	// This can't currently be called at calling class creation since don't have the properties file available at
+	// that time. Each method using WAPI needs to call this.
+
+	public WAPI assureWAPI(WAPI wapi, HashMap<String,String> ioproperties) {
+
+		// Set reuseConnection value from properties.  It is done here to keep changes to a minimum and
+		// because the script is plenty fast.
+
+		// If not set in the class then default value and override from properties (if specified).
+		if (this.reuseConnection == null) {
+			this.reuseConnection = Boolean.valueOf("TRUE");
+			if (ioproperties.get(REUSE_CONNECTION) != null) {
+				this.reuseConnection = Boolean.valueOf(ioproperties.get(REUSE_CONNECTION));
+			}
+		}
+
+		if (this.reuseConnection == false) {
+			wapi = null;
+		}
+
+		// Don't check here for problems, allow any problem to be dealt with in the call chain.
+		if (wapi == null) {
+			wapi = new WAPI(ioproperties);
+		}
```

Comment:
will there be a case `reuseConnection ==true` and `wapi == null`? I just ask b'cause I see 2 different statement could be merged if above case don't happen

Reply:
That case can happen.  The checks can't be merged.